### PR TITLE
GUVNOR-2572 : Refreshing browser page when in authoring > Repository view causes error

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/GUVNOR-2572

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-backend/src/test/java/org/kie/workbench/common/screens/explorer/backend/server/HelperWrapper.java [hunk 1]
```diff
@@ -36,12 +36,17 @@
...
     private ExplorerServiceHelper helper;
+    private boolean includePackage = true;
 
 
     public HelperWrapper(ExplorerServiceHelper helper) {
         this.helper = helper;
     }
 
+    public void excludePackage() {
+        includePackage = false;
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-backend/src/test/java/org/kie/workbench/common/screens/explorer/backend/server/HelperWrapper.java]: What's the purpose of this magic setting? Is it for your unit tests only? Would a different visibility make more sense? package? protected?

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-backend/src/test/java/org/kie/workbench/common/screens/explorer/backend/server/ProjectExplorerContentResolverTest.java [hunk 2]
```diff
@@ -248,6 +248,38 @@ public void testChangeProjectOnTechnicalView() {
         assertNull( content.getSelectedPackage() );
     }
 
+    @Test
+    public void testChangeProjectOnTechnicalView02() {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-backend/src/test/java/org/kie/workbench/common/screens/explorer/backend/server/ProjectExplorerContentResolverTest.java]: I know @csadilek is keen to promote meaningful test names.. even if they become quite verbose (but TBH they do help *later* when you revisit code to ponder what it was actually testing for).

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-backend/src/test/java/org/kie/workbench/common/screens/explorer/backend/server/HelperWrapper.java (hunk 1 above)
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-backend/src/test/java/org/kie/workbench/common/screens/explorer/backend/server/HelperWrapper.java]: It is just for test setup. The class is only used when testing. 

Based on the quick look I noticed the test coverage is poor. There should be also a setting for includeFolderItem and probably even more. I was tempted to test the class properly, but this fix and the V&V are needed this week. I'll make it better next time.

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-backend/src/test/java/org/kie/workbench/common/screens/explorer/backend/server/ProjectExplorerContentResolverTest.java (hunk 2 above)
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-backend/src/test/java/org/kie/workbench/common/screens/explorer/backend/server/ProjectExplorerContentResolverTest.java]: It is a good practice. Copy pasted the one on the top in a hurry. I'll make the name better.

**Comment by Rikkola** [comment]
Fixed the test name.
Jenkins, please retest this.

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-backend/src/test/java/org/kie/workbench/common/screens/explorer/backend/server/HelperWrapper.java (hunk 1 above)
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-backend/src/test/java/org/kie/workbench/common/screens/explorer/backend/server/HelperWrapper.java]: OOps, my bad.

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-backend/src/test/java/org/kie/workbench/common/screens/explorer/backend/server/ProjectExplorerContentResolverTest.java [hunk 3]
```diff
@@ -248,6 +248,38 @@ public void testChangeProjectOnTechnicalView() {
         assertNull( content.getSelectedPackage() );
     }
 
+    @Test
+    public void testChangeProjectOnTechnicalViewWhenThereIsAFolderItemButNoActiveProjectOrganizationalUnitOrRepository() {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-backend/src/test/java/org/kie/workbench/common/screens/explorer/backend/server/ProjectExplorerContentResolverTest.java]: Liking the test name. I think it "top trumps" my previous best.. @triceo What do you think? ;-)

**Comment by manstis** [comment]
+1 to merge.

**Pull request merged**